home/import_helpers.py

```diff
@@ -157,3 +163,55 @@ def extract_errors(data: dict[str | int, Any] | list[str]) -> dict[str, str]:
         pass
 
     return error_message
+
+
+def fix_rows(rows: Generator[dict[str, str], None, None]) -> Iterator[dict[str, str]]:
+    """
+    Fix keys for all rows by lowercasing and removing whitespace
+    """
+    for row in rows:
+        yield fix_row(row)
+
+
+def fix_row(row: dict[str, str]) -> dict[str, str]:
+    return {k.lower().strip(): v.strip() if v else v for k, v in row.items()}
+
+
+def parse_file(
+    file_content: bytes, file_type: str
+) -> Iterator[tuple[int, dict[str, Any]]]:
+    read_rows = read_xlsx if file_type == "XLSX" else read_csv
+    return enumerate(fix_rows(read_rows(file_content)), start=2)
+
+
+def read_csv(file_content: bytes) -> Generator[dict[str, Any], None, None]:
+    try:
+        for row in csv.DictReader(StringIO(file_content.decode())):
+            yield {k.strip().lower(): v for k, v in row.items()}
```

Comment:
Are we not doing this in `fix_rows`? Maybe we can just return the DictReader here?

Reply:
read_csv and read_excel need to both return a Generator though

Reply:
Looks like I can wrangle the types and turn read_csv into a one liner